velox/functions/prestosql/tests/utils/FunctionBaseTest.h

```diff
@@ -250,6 +250,114 @@ class FunctionBaseTest : public testing::Test,
   static std::unordered_set<std::string> getSignatureStrings(
       const std::string& functionName);
 
+  /// Given an expression, a list of inputs and expected results, generate
+  /// dictionary-encoded and constant-encoded vectors, evaluate the expression
+  /// and verify the results.
+  template <typename TInput, typename TOutput>
+  void testEncodings(
+      const std::vector<VectorPtr>& inputs,
+      const core::TypedExprPtr& expr,
+      const VectorPtr& expected) {
+    if (inputs.size() == 0) {
+      return;
+    }
+    auto testDictionary =
+        [&](bool addNulls,
```

Comment:
Consider passing 3 arguments: dictionary size, a lambda to generate indices, an optional lambda to generate nulls.